### PR TITLE
OSASINFRA-3182: openstack: remove portTargets

### PR DESCRIPTION
**Pull request by EmilienM**
The FailureDomain implementation in 4.13 was experimental and delivered
as TechPreview.

In 4.14, we'll deliver a stable FailureDomain API for OpenStack that
will be used in CPMS.

The `porttargets` will have to be re-worked so we can't continue to have
it in the installer since its interface is very likely going to change
for something else in the future.

So we decided to remove it so we can switch the FailureDomain to use the
stable API that was merged in 4.14.

### CODE REVIEW

File: data/data/install.openshift.io_installconfigs.yaml
Comment (EmilienM) [inline data/data/install.openshift.io_installconfigs.yaml:1]: I didn't remove `failureDomains` from this file yet.

Comment (pierreprinetti) [reply data/data/install.openshift.io_installconfigs.yaml:1]: it's done

File: pkg/tfvars/openstack/ports.go [hunk 1]
```diff
@@ -21,85 +9,3 @@ type terraformPort struct {
 	NetworkID string             `json:"network_id"`
 	FixedIP   []terraformFixedIP `json:"fixed_ips"`
 }
-
-func portTargetToTerraformPort(networkClient *gophercloud.ServiceClient, portTarget types_openstack.PortTarget) (terraformPort, error) {
```
Comment (MaysaMacedo) [inline pkg/tfvars/openstack/ports.go:25]: This is being used for the dual-stack support:
https://github.com/openshift/installer/pull/6797/files#diff-069c83ecc685bc97ef280d65021909c03ea67f51f67a998bb3a5cb647b8cf16bR183

File: pkg/types/openstack/machinepool.go [hunk 2]
```diff
@@ -95,56 +85,3 @@ type RootVolume struct {
...
-// intended purpose. If ID is set to "control-plane", then the PortTarget will
-// replace the default cluster primary network (or the machinesSubnet if
-// defined) as the first network for the machine.
-type NamedPortTarget struct {
-	ID         string `json:"id"`
-	PortTarget `json:",inline"`
-}
-
-// PortTarget defines, directly or indirectly, one or more subnets where to attach a port.
-type PortTarget struct {
```
Comment (MaysaMacedo) [inline pkg/types/openstack/machinepool.go:132]: PortTarget is being used for the dual-stack support:
https://github.com/openshift/installer/pull/6797/files#diff-0b0e430b5843d7e62d8138813bc0517be9584d7c77e3ad30686b3fc3dde02e21L132-L138

**Comment by EmilienM** [comment]
@patrickdillon before requesting it to be in the OpenShift 4.14 Feature Merge Queue, could you or someone else in your team please look at this patch and approve it if it's fine.
Thanks

File: pkg/types/openstack/machinepool.go (hunk 2 above)
Comment (pierreprinetti) [reply pkg/types/openstack/machinepool.go:132]: Or even better if we can merge your PR first

**Comment by patrickdillon** [comment]
/approve

nit: it would be best to separate vendoring into a separate commit. fwiw I have found the pattern described in [dependencies.md](https://github.com/openshift/installer/blob/master/docs/dev/dependencies.md), go.(mod,sum) in a separate commit, to be personally very useful.

**Comment by pierreprinetti** [comment]
> /approve
> 
> nit: it would be best to separate vendoring into a separate commit. fwiw I have found the pattern described in [dependencies.md](https://github.com/openshift/installer/blob/master/docs/dev/dependencies.md), go.(mod,sum) in a separate commit, to be personally very useful.

So that would be
* one commit for go.mod, go.sum and vendor/; OR
* one commit for go.{mod,sum} and another for vendor/
?

Where it gets complicated, is when a dep bump requires you to update some function call. Would you commit that together with the bump? I personally would.

However I understand that this PR is the other way around: a change in production code depends on a bump, and not the opposite.

**Comment by patrickdillon** [comment]
This: 
> one commit for go.{mod,sum} and another for vendor/

Typically we separate the vendor code to make it easier to review. Having go.(mod,sum) in a separate commit makes it easier to review those changes, and I have also found it easier for my development workflows

> Where it gets complicated, is when a dep bump requires you to update some function call. Would you commit that together with the bump? I personally would.

No. I would expect most vendoring to have accompanying code changes, but the general guidelines are to separate all vendoring and I have not seen any issues with that.

**Comment by pierreprinetti** [comment]
> This:
> 
> > one commit for go.{mod,sum} and another for vendor/
> 
> Typically we separate the vendor code to make it easier to review. Having go.(mod,sum) in a separate commit makes it easier to review those changes, and I have also found it easier for my development workflows
> 
> > Where it gets complicated, is when a dep bump requires you to update some function call. Would you commit that together with the bump? I personally would.
> 
> No. I would expect most vendoring to have accompanying code changes, but the general guidelines are to separate all vendoring and I have not seen any issues with that.

OK! I'll try to comply. Thank you for the explanation!

**Pull request merged**